code/game/mecha/mech_fabricator.dm

```diff
@@ -182,7 +182,11 @@
 	desc = initial(desc)
 
 	var/obj/item/I = new D.build_path
-	I.loc = get_step(src,SOUTH)
+	if(D.locked)
+		var/obj/item/weapon/storage/lockbox/large/L = new /obj/item/weapon/storage/lockbox/large(src.loc) //(Don't use capitals in paths, or single letters.

```

Comment:
you're placing it on top of the machine; this isn't standard behavior in line with everything else produced by the mech fab).